index.bs

```diff
@@ -654,13 +730,17 @@ support for example recursive iteration.
 The [=asynchronous iterator initialization steps=] for a {{FileSystemDirectoryHandle}} |handle|
 and its async iterator |iterator| are:
 
-1. Let |entry| be the result of [=locating an entry=]
-   given |handle|'s [=FileSystemHandle/locator=].
-1. Let |access| be the result of running |entry|'s
-   [=file system entry/query access=] given "`read`".
-
-1. If |access| is not "{{PermissionState/granted}}",
-   [=throw=] a "{{NotAllowedError}}" {{DOMException}}.
+1. [=Enqueue the following steps=] to the [=file system queue=]:
+  1. Let |entry| be the result of [=locating an entry=]
+     given |handle|'s [=FileSystemHandle/locator=].
+  1. Let |accessResult| be the result of running |entry|'s
+     [=file system entry/query access=] given "`read`".
+  1. If |accessResult|'s [=file system access result/permission state=]
+     is not "{{PermissionState/granted}}":
+    1. Set |accessErrorName| to |accessResult|'s
+       [=file system access result/error name=] if it is not
+       « the empty string »; otherwise "{{NotAllowedError}}".
+    1. [=Throw=] an |accessErrorName| {{DOMException}}.
```

Comment:
We've specified that `query access` requires going in parallel on the `file system queue`, but if we get an exception there's no associated promise to reject...

... @mkruisselbrink do you know if this is this even necessary? Or is it enough to check `query access` in just the "get the next iteration result" algorithm below?

Reply:
It doesn't seem to do anything based on these steps. Per https://webidl.spec.whatwg.org/#idl-async-iterable we can omit it I think.

cc @domenic

Reply:
Tentatively omitted it for now, pending input from others